benchmarks/HOTPOTQA/hotpotqa_benchmark.py

```diff
@@ -0,0 +1,83 @@
+import argparse
+import os
+import time
+from typing import Dict, Tuple, Any
+
+from datasets import Dataset
+
+from benchmarks import BaseBenchmark
+from rageval.metrics import (AnswerEMCorrectness, AnswerF1Correctness)
+
+
+class HOTPOTQABenchmark(BaseBenchmark):
+    """Benchmark for HotPotQA dataset.
+
+    HotPotQA is a new dataset with 113k Wikipedia-based question-answer pairs with four key features: (1) the questions require finding and reasoning over multiple supporting documents to answer; (2) the questions are diverse and not constrained to any pre-existing knowledge bases or knowledge schemas; (3) we provide sentence-level supporting facts required for reasoning, allowingQA systems to reason with strong supervision and explain the predictions; (4) we offer a new type of factoid comparison questions to test QA systems’ ability to extract relevant facts and perform necessary comparison.
+
+    """
+
+    name = "hotpot_qa_benchmark"
+
+    def __init__(self) -> None:
+        """Initialization."""
+        super().__init__()
+
+    def _evaluate(self, ) -> Tuple[Dict[Any, Any], Dataset]:
+        """Evaluate the dataset and return the dataset with scores.
+
+        For the HotPotQA dataset, the `short_answers` and `long_answers` which we need to extract.
+
+
+        We use the `answer` as the `gt_answers` to evaluate the string Exact Match correctness and the `gt_supporting_facts` to evaluate the F1.
+        """
+        self.metrics = [AnswerEMCorrectness(ignore_case=False),
+                        AnswerF1Correctness()
+                        ]
+        ground_truths = {
+            "answer_f1": ("supporting_answer", "gt_supporting_facts"),
```

Comment:
The usage of the F1 metric is incorrect. Please take another look at the code of `AnswerF1Correctness()`. 

When calculating the F1, the two input strings are split into word-level tokens for computation. In this benchmark, “gt_supporting_facts” is a list of supporting sentences, while “supporting_answer” is a string concatenated from the supporting sentences found by the model. Directly inputting “supporting_answer” and “gt_supporting_facts” into `AnswerF1Correctness()` will first calculate the F1 between the token set of the former and the token set of each sentence in the latter, and then take the maximum value from all the F1 scores.
The correct way is calculating the F1 score at sentence level, comparing sentences and their `sent_id` to  determine whether the model has found the appropriate sentences.